tests/lax_numpy_test.py

```diff
@@ -5962,6 +5972,46 @@ def np_fun(a, i, v):
     self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker)
     self._CompileAndCheck(jnp_fun, args_maker)
 
+  @jtu.sample_product(
+    [
+      dict(a_shape=a_shape, i_shape=i_shape, v_shape=v_shape, axis=axis)
+      for a_shape in array_shapes
+      for axis in list(range(-len(a_shape), len(a_shape)))
+      if a_shape[axis] > 0
+      for i_shape in [tuple_replace(a_shape, axis, J) for J in range(5)]
+      for v_shape in broadcastable_shapes(i_shape)
+    ] + [
+      dict(a_shape=a_shape, i_shape=i_shape, v_shape=v_shape, axis=None)
+      for a_shape in nonempty_array_shapes
+      for i_shape in [(J,) for J in range(5)]
+      for v_shape in broadcastable_shapes(i_shape)
+    ],
```

Comment:
Thanks for updating – still, this generates a list of ~1500 test cases, from which we'll only ever select a handful (defined by `JAX_NUM_GENERATED_CASES` which defaults to 10, and goes as high as 90 in some of our beefier CI jobs).

So generating a list of 1500 cases here is wasteful. Maybe it would be better to explicitly list a dozen or so combinations that cover relevant cases?